### PR TITLE
Add bridge hub and statemint/e to e2e environment

### PR DESCRIPTION
**Pull request by alistair-singh**
Related PR to make bridgehub the main parachain with the beacon client included in its runtime. https://github.com/Snowfork/cumulus/pull/2
Clone Snowfork cumulus repo and run...
```cmd
git clone git@github.com:Snowfork/cumulus.git
git checkout alistair/add-snowbridge-pallets
(cd cumulus && cargo build --release --bin polkadot-parachain)
```
I use vscode, I have added the `cumulus` repo to my snowbridge workspace so I can work on both together.
Then set `CUMULUS_BIN` in your `core/packages/test/.envrc` to the build `polkadot-parachain` binary.
```sh
export CUMULUS_BIN=/path/to/cumulus/target/release/polkadot-parachain
```
And then run in the `core/packages/test` folder:
```cmdline
./scripts/start-services.sh
```

# TODO
- [x] Update readme files

### CODE REVIEW

File: core/packages/test/config/launch-config.toml [hunk 1]
```diff
@@ -41,27 +41,92 @@ default_args = [
     name = "rococo-dave"
     validator = true
 
-# Snowbridge
+## Snowbridge
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:44]: Rather just delete commented out code. We can always pull it from the version history

**Review by vgeddes — COMMENTED** [review]
In a future PR we also need to remove the snowbridge parachain. Basically everything except for the following dirs:
  * parachain/pallets
  * parachain/primitives
  * parachain/tools
  * parachain/templates
  * parachain/scripts

And I think even from that list we can cull a few items in `parachain/templates` and `parachain/scripts`. Or perhaps those items need to be moved to cumulus repo.

**Comment by alistair-singh** [comment]
> * parachain/templates

I have a TODO to regenerate weights in the cumulus repo. I will create a ticket for the removal of our runtimes.

I am going to leave this PR open until i have done a bit more testing.

**Comment by yrong** [comment]
are we gonna also to add our [basic_channel_rpc](https://github.com/Snowfork/snowbridge/blob/be58054456a890b46574d5a8483b520ac11b09ef/parachain/src/rpc.rs#L63-L69) into [cumulus](https://github.com/Snowfork/cumulus/blob/21141f7a43ad92bd779d078e308962ee80476787/polkadot-parachain/src/rpc.rs#L71-L73) ?

**Comment by alistair-singh** [comment]
> Are we gonna also to add [basic_channel_rpc](https://github.com/Snowfork/snowbridge/blob/be58054456a890b46574d5a8483b520ac11b09ef/parachain/src/rpc.rs#L63-L69) into [cumulus](https://github.com/Snowfork/cumulus/blob/21141f7a43ad92bd779d078e308962ee80476787/polkadot-parachain/src/rpc.rs#L71-L73) ?

Yes good point. I will add this as well! I didn't see anything fail because we are not actually sending a message across the bridge.

**Comment by alistair-singh** [comment]
> Are we gonna also to add [basic_channel_rpc](https://github.com/Snowfork/snowbridge/blob/be58054456a890b46574d5a8483b520ac11b09ef/parachain/src/rpc.rs#L63-L69) into [cumulus](https://github.com/Snowfork/cumulus/blob/21141f7a43ad92bd779d078e308962ee80476787/polkadot-parachain/src/rpc.rs#L71-L73) ?

Addressed in https://github.com/Snowfork/cumulus/pull/2/commits/5180e63c6710a9a709e50fd4d6655defdddfd352.

**Comment by alistair-singh** [comment]
This can be merged once https://github.com/Snowfork/cumulus/pull/2 is merged.

File: core/packages/test/config/launch-config.toml [hunk 2]
```diff
@@ -41,27 +41,67 @@ default_args = [
     name = "rococo-dave"
     validator = true
 
-# Snowbridge
+## Bridge Hub
 [[parachains]]
+id = 1013
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:46]: Should use the same ID as BridgeHub on Rococo, to minimize configuration overhead.

```suggestion
id = 1002
```

File: core/packages/test/config/launch-config.toml [hunk 3]
```diff
@@ -41,27 +41,67 @@ default_args = [
...
-    command = "{{output_bin_dir}}/snowbridge"
-    name = "snowbrige-bob"
+    name = "bh-bob"
     validator = true
+    command = "{{output_bin_dir}}/polkadot-parachain"
     args = ["--enable-offchain-indexing=true","--pruning=archive"]
+
+## Statemine
+[[parachains]]
+id = 2000
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:73]: Should use the same ID as Statemint on Rococo, to minimize configuration overhead.
```suggestion
id = 1000
```

File: core/packages/test/config/launch-config.toml [hunk 4]
```diff
@@ -41,27 +41,67 @@ default_args = [
...
+chain = "statemine-local"
+cumulus_based = true
+
+    # run alice as parachain collator
+    [[parachains.collators]]
+    name = "statemine-alice"
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:79]: I prefer to move over to this naming:

```suggestion
    name = "statemine01"
```

File: core/packages/test/config/launch-config.toml [hunk 5]
```diff
@@ -41,27 +41,67 @@ default_args = [
...
+    rpc_port = 8082
+    ws_port = 12144
+
+        [[parachains.collators.env]]
+        name = "RUST_LOG"
+        value = "runtime=debug,parachain=trace,cumulus-collator=trace,aura=trace,xcm=trace"
+
+    # run bob as parachain collator
+    [[parachains.collators]]
+    name = "statemine-bob"
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:91]:
```suggestion
    name = "statemine02"
```

File: core/packages/test/config/launch-config.toml [hunk 6]
```diff
@@ -41,27 +41,67 @@ default_args = [
...
 chain_spec_path = "{{output_dir}}/spec.json"
 cumulus_based = true
-id = 1000
 
+    # run alice as parachain collator
     [[parachains.collators]]
-    command = "{{output_bin_dir}}/snowbridge"
-    name = "snowbrige-alice"
-    rpc_port = 8081
+    name = "bh-alice"
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:53]:
```suggestion
    name = "bridgehub01"
```

File: core/packages/test/config/launch-config.toml [hunk 7]
```diff
@@ -41,27 +41,67 @@ default_args = [
...
 
         [[parachains.collators.env]]
         name = "RUST_LOG"
         value = "runtime=debug,parachain=trace,cumulus-collator=trace,aura=trace,xcm=trace,ethereum-beacon-client=trace"
 
+    # run bob as parachain collator
     [[parachains.collators]]
-    command = "{{output_bin_dir}}/snowbridge"
-    name = "snowbrige-bob"
+    name = "bh-bob"
```
Comment (vgeddes) [inline core/packages/test/config/launch-config.toml:66]:
```suggestion
    name = "bridgehub02"
```

File: core/packages/test/config/launch-config.toml (hunk 2 above)
Comment (alistair-singh) [reply core/packages/test/config/launch-config.toml:46]: Do we not want to mimic the real-life parachain id's?

File: core/packages/test/config/launch-config.toml (hunk 3 above)
Comment (alistair-singh) [reply core/packages/test/config/launch-config.toml:73]: Ditto: Do we not want to mimic the real-life parachain id's?

File: core/packages/test/config/launch-config.toml (hunk 2 above)
Comment (alistair-singh) [reply core/packages/test/config/launch-config.toml:46]: Rococo uses `1013` for BridgeHub. All other environments use `1002`. I am going to leave this as 1013.
```
Polkadot -> Statemint(1000), BridgeHub(Not deployed)
Kusama -> Statemine(1000), BridgeHub(1002)
Rococo -> Rockmine(1000), BridgeHub(1013)
Westend -> Westmint(1000), BridgeHub(1002)
```

**Review by vgeddes — APPROVED** [review]
Looks good 🙌

**Pull request merged**